tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java

```diff
@@ -1333,11 +1341,15 @@ public static ExitDetails run(CompileResult compileResult, String... args) {
 
         try {
             final List<String> actualArgs = new ArrayList<>();
-            actualArgs.add(0, "java");
-            actualArgs.add(1, "-cp");
+            int index = 0;
+            actualArgs.add(index, "java");
+            while (index++ < javaOpts.size()) {
+                actualArgs.add(index, javaOpts.get(index - 1));
+            }
```

Comment:
This is good 🙂. One small change that would look nice while we are at it.
```suggestion
            int index = 0;
            actualArgs.add(index++, "java");
            while (index < javaOpts.size()) {
                actualArgs.add(index++, javaOpts.get(index - 1));
            }
```

Reply:
Shall we also use a different index for `javaOpts`? Then we can bring back the for loop:
```suggestion
            int index = 0;
            actualArgs.add(index++, "java");
            for (int i = 0; i < javaOpts.size(); i++) {
                actualArgs.add(index++, javaOpts.get(i));
            }
```
This is because to allow code to be easily changed, when someone wants to add another value to the `actualArgs` before the loop.

Reply:
I will look into these.